File: Frontend/src/types/documents.ts
import type { DocumentType, ExtractionResult } from "./extraction";

export interface DocumentSummary {
  id: number;
  image_basename: string;
  created_at: string;
  ocr_model: string;
  status: "extracted" | "ocr_only";
  document_type: DocumentType | null;
  number: string | null;
  municipality: string | null;
}

export interface DocumentList {
  items: DocumentSummary[];
  total: number;
  limit: number;
  offset: number;
}

export interface DocumentDetail extends DocumentSummary {
  extraction: ExtractionResult | null;
  ocr: {
    content: string;
    model_version: string;
    duration_ms: number;
    warnings: string[];
    regions: {
      id: string;
      text: string | null;
      page: number;
      kind: string | null;
      bounding_box: number[] | null;
    }[];
  };
}

File: app/schemas/documents.py
from typing import Literal

from pydantic import BaseModel

from app.schemas.extraction import ExtractionResult
from app.schemas.ocr import OCRResult


class DocumentSummary(BaseModel):
    id: int
    image_basename: str
    created_at: str
    ocr_model: str
    status: Literal["extracted", "ocr_only"]
    document_type: str | None = None
    number: str | None = None
    municipality: str | None = None


class DocumentList(BaseModel):
    items: list[DocumentSummary]
    total: int
    limit: int
    offset: int


class DocumentDetail(DocumentSummary):
    ocr: OCRResult
    extraction: ExtractionResult | None
